src/lib/components/composites/project-components/CreateProjectDialog.svelte

```diff
@@ -0,0 +1,280 @@
+     * the members with the best matching name or email are at the beginning of the list (and will
+     * appear at the top of the suggestions list).
+     *
+     * @param input the content of the input field, i.e. the search string
+     * @returns list of "name \<email\>" (sorted) representations of users that can be invited
+     */
+    function filterPossibleMembers(input: string): string[] {
+        const fzf = new Fzf(possibleMembers.map((user) => `${getName(user)} <${user.email}>`));
+        return fzf.find(input.toLowerCase()).map((result) => result.item);
+    }
+
+    /**
+     * Checks, whether a given input is a valid name of a registered user or an email.
+     */
+    function validateInput(input: string): ValidationResult {
+        if (!Schema.email.safeParse(input.trim()).success) {
+            const matchingMembers = possibleMembers.filter(
+                (member) => getName(member) === input.trim(),
+            );
+            if (matchingMembers.length === 0) {
+                return { success: false, error: "Please enter a valid name or email." };
+            }
+            if (matchingMembers.length > 1) {
+                return {
+                    success: false,
+                    error: "There are multiple users with this name. Please specify the user.",
+                };
+            }
+        }
+        return { success: true };
+    }
+
+    /**
+     * Maps a valid name to the email of the user.
+     *
+     * If multiple users with the given name exist, the name can not be mapped and a hint is displayed.
+     *
+     * @example The user {firstName: "John", lastName: "Doe, email: "john.doe@example.com", ...}
+     * is in the list of possible members. Then the input
+     * - "John Doe"
+     * - "john.doe@example.com"
+     * - "John Doe <john.doe@example.com>"
+     * all are mapped to "john.doe@example.com"
+     *
+     * @param input the name, email or combination of name \<email\> of a known user
+     * @returns the corresponding email of the user identified by the given name, email or name + email combination
+     */
+    function mapNameToEmail(input: string): string | undefined {
+        let possibleMatchedUser = possibleMembers.filter(
+            (user) =>
+                getName(user) === input ||
+                user.email === input ||
+                `${getName(user)} <${user.email}>` === input,
+        );
+
+        return possibleMatchedUser !== undefined ? possibleMatchedUser.at(0)?.email : undefined;
+    }
+
+    /**
+     * Get the name of a user identified by its email.
+     *
+     * @param input the email of the user
+     * @return the name of the user or undefined, if the no user with the given email was found
+     */
+    function mapEmailToName(input: string): string | undefined {
+        let name = getNames(initialPossibleMembers.filter((user) => input === user.email));
+        return name !== "" ? name : undefined;
```

Comment:
Here, we have to return the input, i.e., the email, when it couldn't be resolved to a name. This was introduced by setting the default value of `displayItem` to the ID function. Earlier we caught that case.
```suggestion
        return name !== "" ? name : input;
```

Reply:
I will change it